chart/kubeapps/templates/dashboard-config.yaml

```diff
@@ -50,6 +54,8 @@ data:
     }
   custom_style.css: |-
 {{- .Values.dashboard.customStyle | nindent 4 }}
+  custom_components.js: |-
+{{- .Files.Get .Values.dashboard.customComponents | nindent 4 }}
```

Comment:
But dont you think copying the file into the chart directory is still much easier than copying the content of the file into the values file? The reason we really like this approach is so that developers can easily hook their custom components into their own CI/CD for deploying kubeapps as a dependency in their own chart

Reply:
Not really, in that case you need to pull and unpack the chart and then copy the file before you are able to install it. Note that you don't need to manually copy the file. In your use case, you can simply do:

```
helm install bitnami/kubeapps --set-file dashboard.customComponents=./docs/developer/examples/CustomComponent.min.js <other_flags>
```

That would read and properly set the file from any path.

Reply:
ohhhhh okay I see, this seems way better. will update and add some docs 👍